usr/share/rear/setup/default/005_ssh_agent_start.sh

```diff
@@ -0,0 +1,11 @@
+# If available, start ssh-agent so that SSH passphrases must be entered only once during 'rear recover' even with
+# multiple ssh invocations (as may be the case when restoring from a network backup server).
+#
+# This file is part of Relax-and-Recover, licensed under the GNU General
+# Public License. Refer to the included COPYING for full text of license.
+
+if has_binary ssh-agent; then
+    Log "Starting up ssh-agent"
+    eval "$(ssh-agent -s)"
+    echo -e "\nHost *\nAddKeysToAgent yes\n" >> /root/.ssh/config
```

Comment:
According to https://www.openssh.com/txt/release-7.2 this option was added in openssh 7.2 in February 2016. Are you sure that all supported distros have this or a newer version? What happens with older ssh? Why not call `ssh-add` to add all the keys?

(Thanks for pointing me to this option, I didn't know about it)

Another thought: Should we indicate to the user that the SSH key passphrase was changed or set?

Reply:
1. Actually I'm not aware about ssh versions on the other distros. Hopefully others could point out some details.
1. An older ssh might just ignore the option, but I don't know für sure. Should probably check this.
1. I was thinking about using `ssh-add` first but wanted to ask the user for a password only whenever it is actually used for an ssh call. After all, we cannot not know a priori which ssh invocations require which private key, if at all.
1. Regarding indications on setting/changing passphrases: Currently, the passphrase question will come up only if required. So the user knows that she would have to remember the passphrase just entered. The user will also be informed by `LogUserOutput "NOTE: ...` if another previously protected private key (with a potentially different passphrase) exists. The user will then find complete details in the log. Anything missing here?

Reply:
First of all, I misunderstood your code. You are right, the agent only runs in the context of a ReaR flow. The "right" way to kill the agent is with the help of `AddExitTask` which you set right after successfully starting the agent.

I imagine that Ubuntu 14, CentOS 5 or SLES 11 will be a good test case for older SSH, please check that this solution also works with older SSH.

I like your goal of only asking passphrases for keys that are actually used. We might have to accept the fact that this goal only works on newer systems. For older systems we still have to provide a useful user experience.

"just entered" only applies to us ReaR developers. Most users employ ReaR as an "install & forget" solution where they automate the mkrescue and don't remember much from the ReaR setup when they actually use the recovery media. That's why I think some guidance will be helpful here.